File: stripe/php/stripe_client.php
<?php
/**
 * Dreamcobots Stripe Integration — PHP
 *
 * Install: composer install
 * Configure .env (see ../../.env.example)
 */

require_once __DIR__ . '/vendor/autoload.php';

$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->safeLoad();

\Stripe\Stripe::setApiKey($_ENV['STRIPE_SECRET_KEY'] ?? '');

class DreamcobotsStripe
{
    /**
     * Create a Stripe Checkout session.
     */
    public static function createCheckoutSession(
        int $amountCents,
        string $currency,
        string $customerEmail,
        string $successUrl,
        string $cancelUrl,
        string $mode = 'payment'
    ): array {
        $session = \Stripe\Checkout\Session::create([
            'payment_method_types' => ['card'],
            'line_items' => [[
                'price_data' => [
                    'currency'     => strtolower($currency),
                    'product_data' => ['name' => 'Dreamcobots Service'],
                    'unit_amount'  => $amountCents,
                ],
                'quantity' => 1,
            ]],
            'mode'           => $mode,
            'customer_email' => $customerEmail,
            'success_url'    => $successUrl,
            'cancel_url'     => $cancelUrl,
        ]);

        return ['session_id' => $session->id, 'checkout_url' => $session->url];
    }

    /**
     * Create a shareable Payment Link.
     */
    public static function createPaymentLink(
        int $amountCents,
        string $currency,
        string $productName
    ): array {
        $price = \Stripe\Price::create([
            'unit_amount'  => $amountCents,
            'currency'     => strtolower($currency),
            'product_data' => ['name' => $productName],
        ]);
        $link = \Stripe\PaymentLink::create([
            'line_items' => [['price' => $price->id, 'quantity' => 1]],
        ]);
        return ['id' => $link->id, 'url' => $link->url];
    }
}

if (basename(__FILE__) === basename($_SERVER['PHP_SELF'] ?? '')) {
    $mode = !empty($_ENV['STRIPE_SECRET_KEY']) ? 'live' : 'simulation';
    echo "Dreamcobots Stripe PHP client initialised in {$mode} mode.\n";
}

File: stripe/ruby/stripe_client.rb
# frozen_string_literal: true
# Dreamcobots Stripe Integration — Ruby
#
# Install: bundle install
# Configure .env (see ../../.env.example)

require "dotenv/load"
require "stripe"

Stripe.api_key = ENV.fetch("STRIPE_SECRET_KEY", "")

module DreamcobotsStripe
  # Create a checkout session for a one-time payment
  def self.create_checkout_session(amount_cents:, currency: "usd", customer_email:,
                                   success_url:, cancel_url:, mode: "payment")
    session = Stripe::Checkout::Session.create(
      payment_method_types: ["card"],
      line_items: [{
        price_data: {
          currency: currency.downcase,
          product_data: { name: "Dreamcobots Service" },
          unit_amount: amount_cents
        },
        quantity: 1
      }],
      mode: mode,
      customer_email: customer_email,
      success_url: success_url,
      cancel_url: cancel_url
    )
    { session_id: session.id, checkout_url: session.url }
  end

  # Create a shareable Payment Link
  def self.create_payment_link(amount_cents:, currency: "usd", product_name:)
    price = Stripe::Price.create(
      unit_amount: amount_cents,
      currency: currency.downcase,
      product_data: { name: product_name }
    )
    link = Stripe::PaymentLink.create(
      line_items: [{ price: price.id, quantity: 1 }]
    )
    { id: link.id, url: link.url }
  end

  # Create a subscription
  def self.create_subscription(customer_id:, price_id:)
    sub = Stripe::Subscription.create(
      customer: customer_id,
      items: [{ price: price_id }]
    )
    { id: sub.id, status: sub.status }
  end
end

if __FILE__ == $PROGRAM_NAME
  mode = ENV["STRIPE_SECRET_KEY"] ? "live" : "simulation"
  puts "Dreamcobots Stripe Ruby client initialised in #{mode} mode."
end
